lib/curl_req.ex

```diff
@@ -60,6 +59,12 @@ defmodule CurlReq do
         req
       end
 
+    auth =
+      case req.options do
+        %{auth: {:basic, credentials}} -> ["-u", credentials]
```

Comment:
I have been thinking already about doing this kind of processing. Woytek mentioned converting `-H accept_encoding...` to `--compressed` instead of the header version [here](https://x.com/wojtekmach/status/1795673299630141872) but I am hesitant. Why? Because this isn't the request that Req is sending. So I think there are two modes here perhaps. One is "exact" mode, like the request exactly as Req is going to send it (or as close as we can get). And the second is the "approximate" mode (for lack of better terminology), which converts things like basic auth to `-u` and a compression header to `--compressed.

There is then also the question of implementation. The code, as written here, won't work for all cases. Someone could, for instance, add their own auth header, or define a step that does something with the auth option.

I am wondering if we should pull these changes out into a new PR for now. Then we can continue to discuss and refine the hard part but still merge the easy parts for now.

Reply:
Thank you for your thoughts on this. I was struggling on the auth part as you can see because we need to filter the headers based on the type of auth. I like your proposal to remove the auth part for now and find a solution with the exact/approximate solution. 👍